spec/features/users_spec.rb

```diff
@@ -1,90 +1,72 @@
-feature "User management" do
+feature "user management" do
   let(:users_index_path) { polymorphic_path([:users]) }
 
-  context "when an admin user visits the users index" do
-    let!(:admin_user) { login_new_admin_user }
-    subject { admin_user }
+  context "when an admin visits the users index" do
+    let!(:admin) { login_new_admin_user }
+    let!(:users) { create_list(:user, 2) }
+    let!(:unapproved_user) { create(:user, approval_at: nil)}
 
     before { visit users_index_path }
 
     it "displays a table of users" do
       expect(page).to have_table("users_table")
     end
 
-    it "fills the table with the right headers and rows" do
-      check_headers_and_values_on_generic_index_page(
-        "users_table", {
-          "Name" => "Example User",
-          "Created At" => subject.created_at_string,
-          "Approval At" => subject.approval_at_string,
-          "Admin" => "true",
-          "Actions" => "Remove Approval Delete",
-        }
-      )
+    it "sees a table with proper headers and rows" do
+      expect(page).to have_css("#user_#{admin.id}", :text => "#{admin.name}")
+      expect(page).to have_css("#user_#{admin.id}", :text => "#{admin.created_at_string}")
+      expect(page).to have_css("#user_#{admin.id}", :text => "#{admin.approval_at_string}")
+      expect(page).to have_css("#user_#{admin.id}", :text => "Delete")
+      expect(page).to have_css("#user_#{admin.id}", :text => "Yes")
+      expect(page).to_not have_css("#user_#{admin.id}", :text => "No")
     end
 
-    context "and a new non-admin user is added" do
-      subject! { User.create!(name: "Joe") }
-
-      before { visit users_index_path }
+    it "sees a table with proper headers and rows" do
+      expect(page).to have_css("#user_#{users.first.id}", :text => "#{users.first.name}")
+      expect(page).to have_css("#user_#{users.first.id}", :text => "#{users.first.created_at_string}")
+      expect(page).to have_css("#user_#{users.first.id}", :text => "#{users.first.approval_at_string}")
+      expect(page).to have_link('Yes', :href => remove_approval_user_path(users.first.id))
+      expect(page).to_not have_link('No', :href => approve_user_path(users.first.id))
+      expect(page).to have_link('No', :href => toggle_admin_user_path(users.first.id))
+      expect(page).to_not have_link('Yes', :href => toggle_admin_user_path(users.first.id))
+    end
 
-      it "fills the table with that user\"s information too" do
-        check_headers_and_values_on_generic_index_page(
-          "users_table", {
-            "Name" => "Joe",
-            "Created At" => subject.created_at_string,
-            "Approval At" => "",
-            "Admin" => "",
-            "Actions" => "Approve Delete",
-          }
-        )
+    context "the admin clicks Yes for user approval" do
+      before do
+        find(:xpath, "//a[@href='/users/#{unapproved_user.id}/approve']").click
```

Comment:
Ok. Since you already had a dynamic button class for the admin approval button, I went ahead and did the exact same thing for the user approval button. This allows me to find by the button class rather than using XPath.